comp/core/workloadmeta/collectors/internal/remote/generic.go

```diff
@@ -58,7 +58,7 @@ type StreamHandler interface {
 	// NewClient returns a client to connect to a remote gRPC server.
 	NewClient(cc grpc.ClientConnInterface) GrpcClient
 	// HandleResponse handles a response from the remote gRPC server.
-	HandleResponse(response interface{}) ([]workloadmeta.CollectorEvent, error)
+	HandleResponse(store workloadmeta.Component, response interface{}) ([]workloadmeta.CollectorEvent, error)
```

Comment:
instead of changing the interface, we could add a field in the StreamHandler struct to have access to the store.

Reply:
Yes 👍 

I was following the same pattern used [here](https://github.com/DataDog/datadog-agent/blob/02c9011e8dad4ba85c8f91eadc1c544eae47eaeb/comp/core/workloadmeta/collectors/internal/remote/generic.go#L63).

If we want to add the store to the stream handler as a field, we would need to modify the `HandleResync` method too.

I don't have a strong opinion about this, both should work and I don't see any privilege (in terms of performance) of one approach compared to the other.

WDYT?

Reply:
make thanks, thanks for the explanations